testQueryHeartbeat is flaky

Another test that looks fundamentally probabilistic and flaky. We really should stop writing and including these.

```
2024-07-17T00:52:18.1917691Z [ERROR] Tests run: 8391, Failures: 1, Errors: 0, Skipped: 1, Time elapsed: 1,346.493 s <<< FAILURE! - in TestSuite
2024-07-17T00:52:18.1920380Z [ERROR] com.facebook.presto.resourcemanager.TestResourceManagerClusterStatusSender.testQueryHeartbeat  Time elapsed: 1.008 s  <<< FAILURE!
2024-07-17T00:52:18.1923186Z java.lang.AssertionError: Expect number of heartbeats to fall within target range (10), +/- 50%.  Was: 4 expected [true] but found [false]
2024-07-17T00:52:18.1942673Z 	at org.testng.Assert.fail(Assert.java:110)
2024-07-17T00:52:18.1943656Z 	at org.testng.Assert.failNotEquals(Assert.java:1413)
2024-07-17T00:52:18.1944462Z 	at org.testng.Assert.assertTrue(Assert.java:56)
2024-07-17T00:52:18.2040828Z 	at 
```

--- a/presto-main/src/test/java/com/facebook/presto/resourcemanager/TestResourceManagerClusterStatusSender.java
+++ b/presto-main/src/test/java/com/facebook/presto/resourcemanager/TestResourceManagerClusterStatusSender.java
@@ -112,7 +112,7 @@ public class TestResourceManagerClusterStatusSender
                 format("Expect number of heartbeats to fall within target range (%s), +/- 50%%.  Was: %s", TARGET_HEARTBEATS, nodeHeartbeats));
     }
 
-    @Test(timeOut = 6_000)
+    @Test(timeOut = 10_000)
     public void testQueryHeartbeat()
             throws Exception
     {
@@ -122,8 +122,14 @@ public class TestResourceManagerClusterStatusSender
         Thread.sleep(SLEEP_DURATION);
 
         int queryHeartbeats = resourceManagerClient.getQueryHeartbeats();
-        assertTrue(queryHeartbeats > TARGET_HEARTBEATS * 0.5 && queryHeartbeats <= TARGET_HEARTBEATS * 1.5,
-                format("Expect number of heartbeats to fall within target range (%s), +/- 50%%.  Was: %s", TARGET_HEARTBEATS, queryHeartbeats));
+        assertTrue(queryHeartbeats > 0, "Expected at least one query heartbeat");
+
+        Thread.sleep(SLEEP_DURATION);
+
+        int newQueryHeartbeats = resourceManagerClient.getQueryHeartbeats();
+        assertTrue(
+                newQueryHeartbeats > queryHeartbeats,
+                format("Expected at least one subsequent query heartbeat, previous: %s, current: %s", queryHeartbeats, newQueryHeartbeats));
 
         // Completing the query stops the heartbeats
         queryExecution.complete();

